### ISSUE TITLE
Unable to apply tags to multiple API operations

### ISSUE THREAD

**Issue by alexwiese** [hashicorp/terraform-provider-azurerm#15921]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v1.1.7
on windows_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.99.0
```

### Affected Resource(s)

* `azurerm_api_management_api_operation_tag`

### Terraform Configuration Files

```hcl
resource "azurerm_api_management_api_operation" "example" {
  operation_id        = "user-delete"
  api_name            = azurerm_api_management_api.api.name
  api_management_name = var.api_management_name
  resource_group_name = var.resource_group_name
  display_name        = "Delete User Operation"
  method              = "DELETE"
  url_template        = "/users"
  description         = "This can only be done by the logged in user."

  response {
    status_code = 200
  }
}

resource "azurerm_api_management_api_operation_tag" "example" {
  api_operation_id = azurerm_api_management_api_operation.example.id
  name             = "example-Tag"
   display_name     = "example-Tag"
}


resource "azurerm_api_management_api_operation" "example2" {
  operation_id        = "user-get"
  api_name            = azurerm_api_management_api.api.name
  api_management_name = var.api_management_name
  resource_group_name = var.resource_group_name
  display_name        = "Get User Operation"
  method              = "GET"
  url_template        = "/users"
  description         = "This can only be done by the logged in user."

  response {
    status_code = 200
  }
}

resource "azurerm_api_management_api_operation_tag" "example2" {
  api_operation_id = azurerm_api_management_api_operation.example.id
  name             = "example-Tag"
  display_name     = "example-Tag"
}

```

### Expected Behaviour

Both operations, `user-delete` and `user-get`, should be tagged with the "example-Tag".

### Actual Behaviour

One operation (`user-get`) gets tagged with "example-Tag".

An error occurs when applying the tag to `user-delete`:
```
Error: A resource with the ID "/subscriptions/xyz/resourceGroups/abc/providers/Microsoft.ApiManagement/service/apim/apis/aaa/operations/user-delete/tags/example-Tag" already exists - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for "azurerm_api_management_api_operation_tag" for more information.
```

### Steps to Reproduce

1. `terraform apply`

### References

* [api_management_api_operation_tag](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/api_management_api_operation_tag)

**Comment by xuzhang3** [comment]
This is a bug,  `azurerm_api_management_api_operation_tag` try get the tags at API level which will get a tag with the same name defined in another API_Operation.  The get should limited to operation level.

**Comment by arkiaconsulting** [comment]
to me this issue is fixed (tested with 3.30.0)

**Comment by arkiaconsulting** [comment]
@xuzhang3 is there any reason why the tag is created at service level, instead of being created at Api level ? In the project I work on, we only have permissions at the Api level, not at the service level.

**Comment by arkiaconsulting** [comment]
> @xuzhang3 is there any reason why the tag is created at service level, instead of being created at Api level ? In the project I work on, we only have permissions at the Api level, not at the service level.

Now I understand: tags does only exist at service level, and can be associated either at Api level, either at Api operation level.

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #16006. As such, I am going to mark this issue as closed.
